plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/natures/DataflowJavaProjectNature.java

```diff
@@ -59,67 +57,32 @@ public void setProject(IProject project) {
    */
   public static void addDataflowJavaNatureToProject(IProject project, IProgressMonitor monitor)
       throws CoreException {
-    if (isProjectAccessible(project)) {
+    Preconditions.checkNotNull(project);
+    if (!project.isAccessible()) {
       throw new CoreException(new Status(Status.WARNING, DataflowCorePlugin.PLUGIN_ID,
-          "Can't add the Dataflow nature to nonexistent or closed project " + project.getName()));
```

Comment:
Previously, `project` could potentially be `null`, and a null project brings execution here. `project.getName()` would then throw an NPE.